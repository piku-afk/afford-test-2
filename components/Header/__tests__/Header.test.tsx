import { render, screen } from '@testing-library/react';
import { Header } from '..';
import userEvent from '@testing-library/user-event';

const user = userEvent.setup();

describe('Header Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

it('renders different component correctly', () => {
  render(<Header />);

  const header = screen.getByText(/sample ecom/i);
  const categoryButton = screen.getByRole('button', {
    name: /all categories/i,
  });
  const profileButton = screen.getByRole('button', {
    name: /profile/i,
  });
  const cartButton = screen.getByRole('button', {
    name: /cart/i,
  });

  expect(header).toBeInTheDocument();
  expect(categoryButton).toBeInTheDocument();
  expect(profileButton).toBeInTheDocument();
  expect(cartButton).toBeInTheDocument();
});

it('opens cart popup when cart button is clicked', async () => {
  render(<Header />);

  const cartButton = screen.getByRole('button', {
    name: /cart/i,
  });
  await user.click(cartButton);

  const cartPopper = screen.getByRole('presentation', { name: 'cart-popup' });
  const closeButton = screen.getByRole('button', {
    name: /close-cart/i,
  });

  expect(cartPopper).toBeInTheDocument();
  expect(cartPopper).toHaveAttribute('id', 'show-cart');
  expect(closeButton).toBeInTheDocument();

  await user.click(closeButton);

  expect(screen.queryByRole('presentation', { name: 'cart-popup' })).toBe(null);
});
});
