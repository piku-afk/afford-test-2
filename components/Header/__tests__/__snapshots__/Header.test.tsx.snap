// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl MuiContainer-disableGutters MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-v8jruj-MuiContainer-root-MuiPaper-root-MuiAppBar-root"
      style="padding-left: 48px; padding-right: 48px;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-8a8g2j-MuiToolbar-root"
        style="padding-left: 0px; padding-right: 0px;"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-7t20ig-MuiTypography-root"
        >
          Sample Ecom
        </p>
        <div
          class="MuiBox-root css-1klfqz3"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorSecondary MuiInputBase-fullWidth MuiInputBase-sizeSmall MuiInputBase-adornedStart MuiInputBase-adornedEnd css-b6n6cb-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart css-hh6b6o-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <button
                aria-haspopup="true"
                class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-12rlaf1-MuiButtonBase-root-MuiButton-root"
                id="category-menu-button"
                tabindex="0"
                type="button"
              >
                 All categories
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <input
              aria-label="search"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-12yjm75-MuiInputBase-input-MuiOutlinedInput-input"
              name="search"
              placeholder="Search for Products, categories..."
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            aria-label="Profile"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PersonOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-describedby="show-cart"
            aria-label="Cart"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1subqga-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="BaseBadge-root MuiBadge-root css-1xwrqzj-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShoppingBagOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 6h-2c0-2.21-1.79-4-4-4S8 3.79 8 6H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-6-2c1.1 0 2 .9 2 2h-4c0-1.1.9-2 2-2zm6 16H6V8h2v2c0 .55.45 1 1 1s1-.45 1-1V8h4v2c0 .55.45 1 1 1s1-.45 1-1V8h2v12z"
                />
              </svg>
              <span
                class="BaseBadge-badge BaseBadge-invisible MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginBottomLeft MuiBadge-anchorOriginBottomLeftRectangular MuiBadge-overlapRectangular css-10aq979-MuiBadge-badge"
              />
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-5x9whe"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        style="padding-left: 48px; padding-right: 48px;"
      >
        <div
          class="css-w4z10b-MuiStack-root"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
